sources/sections/05-date-time-format.adoc

```diff
@@ -111,12 +240,10 @@ time-zone-part = time-zone-char *13(time-zone-char / DIGIT / "-" / "+") ; but no
 time-zone-id   = time-zone-part *("/" time-zone-part)
 time-zone      = "[" time-zone-id "]"
 
-calendar-part  = 3*8(ALPHA / DIGIT)
-calendar-id    = calendar-part *("-" calendar-part)
-calendar       = "[c=" calendar-id "]"
-
-suffix-part    = "[" 1*ALPHA "=" 1*VCHAR "]"
-suffix         = [time-zone] [calendar] *suffix-part
+suffix-key     = 1*alphanum "-" 2*alphanum
+suffix-value   = 1*VCHAR *("-" 1*VCHAR)
```

Comment:
The grammar would be clearer if it were:

suffix-key     = namespace ["-" namespaceKey]
namespace = 1*alphanum
namespaceKey =  1*alphanum

suffix-values   = suffix-value *("-" 1*suffix-value)
suffix-value = 1*VCHAR

Add paragraph:

For BCP47 extensions syntax, 
- case differences are ignored
- the namespace is restricted to single alphanum, corresponding to extension singletons ('x' can be used for a private use extension)
- in addition, for CLDR extensions:
    - there must be a namespaceKey, and it is restricted to 2 alphaNum characters
    - a suffix-value is limited to 3*8 alphanum


Other notes: VCHAR is odd, since it can include "-".  So "---" would be valid suffix-value. Do you want that? Also, do you want the values to be any printable, or just alphanum?